### PR TITLE
To treat HW Trigger Mode

### PR DESCRIPTION
**Pull request by ytakio**
Please allow me propose PR to treat HW Trigger Mode.

- Added argument of trigger mode in ensenso_driver.cpp::configureStreaming() to reset at changing trigger mode.
- Parsed a capture command into trigger and retrieve commands for HW Trigger
mode in EnsensoGrabber::processGrabbing().

### CODE REVIEW

File: src/ensenso_grabber.cpp [hunk 1]
```diff
@@ -581,7 +581,26 @@ void pcl::EnsensoGrabber::processGrabbing ()
...
+            }
+            catch (const NxLibException &ex) {
+                // To ignore timeout exception and others
+            }
+        }
+        if (!running_) {
+            return;
+        }
+
         double timestamp;
```
Comment (fsuarez6) [inline src/ensenso_grabber.cpp:604]: Do we need to perform `cmdTrigger` and `cmdRetrieve`?
According to the ensenso manual `cmdCapture` performs `cmdTrigger` and `cmdRetrieve` operations together: https://www.ensenso.com/manual/cmdtrigger.htm

File: src/ensenso_driver.cpp [hunk 2]
```diff
@@ -308,12 +310,14 @@ class EnsensoDriver
       return true;
     }
 
-    bool configureStreaming(const bool cloud, const bool images)
+    bool configureStreaming(const bool cloud, const bool images, const int trigger_mode)
```
Comment (fsuarez6) [inline src/ensenso_driver.cpp:313]: If the `trigger_mode_` is changed you are forcing the camera to restart (first stop and then restart).
I was thinking that we can do that inside the `CameraParametersCallback` function, what do you think?

Comment (ytakio) [reply src/ensenso_driver.cpp:313]: Of course, it is able to do the same thing in the ``CameraParametersCallback``. But I'm thinking that it is good to make a reset (stop and start) in only one function.
Meanwhile, I've recognized ''configureStreaming'' as to do so after reconfiguration.

And making reset is needed for switching to SW trigger from HW trigger.

File: src/ensenso_grabber.cpp (hunk 1 above)
Comment (ytakio) [reply src/ensenso_grabber.cpp:604]: I want to wait for an HW signal without a timeout. ``cmdCapture`` may fail to detect a signal just after timeout error. 
And ``cmdTrigger`` can wait for HW signal w/o timeout.
This code has been checked at both SW trigger mode and HW trigger mode.
It seems working fine :)

**Comment by fsuarez6** [comment]
@ytakio Let me test this PR in our cameras before merging. 
I'm still thinking that the camera restart should be done inside `CameraParametersCallback`.
I'll propose a PR to [your fork](https://github.com/ytakio/ensenso), so, let's discuss that point there.

**Comment by ytakio** [comment]
@fsuarez6 Thanks a lot.

**Comment by ytakio** [comment]
@fsuarez6 @manbou I've just merged. I'd appreciate it if you would confirm it.

**Pull request merged**